Static finish slide

rwaldron opened the pull request:
I built this off the prompt branch, because there are changes there that I needed here.

ktbee commented on client/components/Scenario/FinishSlide.jsx:
```diff
@@ -0,0 +1,83 @@
+                />
+                <Card centered className={className}>
+                    <Card.Content style={{ flexGrow: '0' }}>
+                        <Card.Header>Great work!</Card.Header>
```
I have a feeling that they'll want to be able to edit this header, but not something we have to address now. 🔮

rwaldron replied:
I meant to remove this when `finish` was ...finished 🤣

rwaldron replied:
Removed.

ktbee commented on client/components/Slide/Components/Text/Editor.jsx:
```diff
@@ -7,24 +7,37 @@ import { type } from './type';
+        const defaultHtml = props.value.html || '';
+        const hasWrapper = /^(<.+>.*<\/.+>)$/gm.test(defaultHtml);
+        const defaultValue = convertFromHTML(
+            hasWrapper ? props.value.html : `<p>${props.value.html}</p>`
```
I can't quite hit this error again for some reason, but I found on legacy content this line was throwing an error because `props.value` didn't exist.

rwaldron replied:
Fixed!

ktbee commented on client/components/Scenario/ContentSlide.jsx:
```diff
@@ -160,6 +160,15 @@ class ContentSlide extends React.Component {
 
+        let skipButtonContent = skipButton;
+
+        if (isLastSlide) {
```
I'm realizing we don't need to the index to determine the last slide with this PR, it's saved on the slide itself! That might be worth refactoring in this PR. This way we can make `Scenario/index.jsx` simpler.

rwaldron replied:
We use the `index` as a `key`, otherwise React complains. When I remove it: 

```
Warning: Each child in a list should have a unique "key" prop.
```

rwaldron commented on client/components/Scenario/FinishSlide.jsx:
```diff
@@ -0,0 +1,83 @@
+                        <Card.Header>Great work!</Card.Header>
+                    </Card.Content>
+                    <Card.Content>
+                        {components.map(({ html: __html }, index) => (
```
Fixed

rwaldron commented:
> I can’t dismiss the dialog box for finishing the scenario by clicking either buttons. I need to refresh the page to move on.

Fixed!

> I can’t load final slide for existing Teacher Moments I made before this PR.

I'm not sure I understand this one, let's go over it together.

> I get a 500 server error in the console when I load http://localhost:3000/editor/new

Can't reproduce, let's go over it together as well

> I can see the finish slide in the slides tab before I save it in the scenario tab. It’s removed once I hit save.

That's definitely a bug! I'll work on this first.

> The preview tab doesn’t show the finish slide, should it?

It should, so this is also a bug. 

Thanks again! I'll ping you when I'm ready for another review :)

rwaldron commented:
> I can see the finish slide in the slides tab before I save it in the scenario tab. It’s removed once I hit save.

Fixed!

> The preview tab doesn’t show the finish slide, should it?

Fixed!

rwaldron commented:
https://www.youtube.com/watch?v=xoBpKNVi5qc

rwaldron commented:
Rebased!

rwaldron commented:
Wahooooo!
